ddtrace/ext/aws.py

```diff
@@ -27,6 +28,33 @@ def unpacking_args(args, args_name, traced_args_list):
     return response
 
 
+def flatten_args(args):
+    """Filter operation arguments based on list of arguments to be traced
+    """
+
+    # unnest structure for argument values that are not simple/atomic
+    def _unnest(t):
```

Comment:
You could potentially have this do unnesting at a deeper level too:

```python
dict(
    params=dict(
        Bucket=dict(
            Name='bucket-name',
        ),
    ),
)
```

Would give us `{'params.Bucket.Name': 'bucket-name'}`

I don't actually think this is necessary right now, just pointing it out.

Reply:
Also, I don't see this closing over anything, so no reason to have it be defined in this function, we can define outside.

There is potential for this to be a centrally located util function, I could see this being useful for other integrations as well.


If we do centralize it we probably want the logic to work to flatten as deep as the dicts go, but maybe tax an optional `max_depth` argument or something.


e.g.

```python
def flatten_dict(value, max_depth=None):
    # implementation left as an exercise to the reader
    pass


>>> flatten_dict(dict(params=dict(Bucket=dict(Name='bucket-name'))))
{'params.Bucket.Name': 'bucket-name'}

# Should `max_depth=1` mean `params.Bucket` or `max_depth=2`?
>>> flatten_dict(dict(params=dict(Bucket=dict(Name='bucket-name'))), max_depth=2)
{'params.Bucket': 'bucket-name'}
```


What do you think?

Reply:
Added it to `ddtrace.utils.formats` for now since `asbool` was there and it made sense to keep those two together.